analyzers/conversion_overflow.go

```diff
@@ -162,6 +148,118 @@ func isStringToIntConversion(instr *ssa.Convert, dstType string) bool {
 	}
 }
 
+func hasExplicitRangeCheck(instr *ssa.Convert, dstType string) bool {
+	dstInt, err := parseIntType(dstType)
+	if err != nil {
+		return false
+	}
+
+	srcInt, err := parseIntType(instr.X.Type().String())
+	if err != nil {
+		return false
+	}
+
+	minBoundChecked := checkSourceMinBound(srcInt, dstInt)
+	maxBoundChecked := checkSourceMaxBound(srcInt, dstInt)
+
+	// If both bounds are already checked, return true
+	if minBoundChecked && maxBoundChecked {
+		return true
+	}
+
+	visitedIfs := make(map[*ssa.If]bool)
+	for _, block := range instr.Parent().Blocks {
+		var minChecked, maxChecked bool
+		for _, blockInstr := range block.Instrs {
+			switch v := blockInstr.(type) {
+			case *ssa.If:
+				minChecked, maxChecked = checkIfForRangeCheck(v, instr, dstInt, visitedIfs)
+			case *ssa.Call:
+				// len(slice) results in an int that is guaranteed >= 0, which
+				// satisfies the lower bound check for int -> uint conversion
+				if v != instr.X {
+					continue
+				}
+				if fn, isBuiltin := v.Call.Value.(*ssa.Builtin); isBuiltin && fn.Name() == "len" && !dstInt.signed {
```

Comment:
Can we also account for `cap`? It's not as widely used as `len`, but should be easy to add.